airbyte-webapp-e2e-tests/cypress/commands/connectorBuilder.ts

```diff
@@ -66,4 +91,54 @@ export const assertMultiPageReadItems = () => {
 
   goToTestPage(3);
   testPanelContains('[]');
-};
\ No newline at end of file
+};
+
+const MAX_NUMBER_OF_PAGES = 5;
+export const assertMaxNumberOfPages = () => {
+  assertHasNumberOfPages(MAX_NUMBER_OF_PAGES)
+};
+
+export const assertHasNumberOfSlices = (numberOfSlices: number) => {
+  cy.get('[data-testid="tag-select-slice"] button').click();
+  cy.get('[data-testid="tag-select-slice"] li').should('have.length', numberOfSlices);
+};
+
+const MAX_NUMBER_OF_SLICES = 5;
+export const assertMaxNumberOfSlices = () => {
+  assertHasNumberOfSlices(MAX_NUMBER_OF_PAGES)
+};
+
+export const assertMaxNumberOfSlicesAndPages = () => {
+  for (var i = 0; i < MAX_NUMBER_OF_SLICES; i++) {
+    cy.get('[data-testid="tag-select-slice"] button').click();
+    cy.get('[data-testid="tag-select-slice"] li').contains("Slice " + i).click();
+    assertMaxNumberOfPages();
+  }
+};
+
+const SCHEMA =  ' {\n' +
+'   "$schema": "http://json-schema.org/schema#",\n' +
+'   "properties": {\n' +
+'     "name": {\n' +
+'       "type": "string"\n' +
+'     }\n' +
+'   },\n' +
+'   "type": "object"\n' +
+' }'
+export const assertSchema = () => {
+  cy.get('[data-testid="tag-tab-detected-schema"]').click();
+  cy.get('pre[class*="SchemaDiffView"]').contains(SCHEMA).should("exist");
+};
+
+const SCHEMA_WITH_MISMATCH = '{{}"$schema": "http://json-schema.org/schema#", "properties": {{}"name": {{}"type": "number"}}, "type": "object"}'
+export const acceptSchemaWithMismatch = () => {
+  cy.get('[data-testid="tag-tab-stream-schema"] button').click();
+  cy.get('textarea').type(SCHEMA_WITH_MISMATCH);
+};
+
+export const assertSchemaMismatch = () => {
+  cy.get('[data-testid="tag-tab-detected-schema"]').click();
+  cy.contains("Detected schema and declared schema are different").should("exist");
+  cy.get('pre[class*="SchemaDiffView"]').contains('-      "type": "number"').should("exist");
```

Comment:
This might be too precise and lead to test flakiness. If we agree with this, I can keep the `cy.contains("Detected schema and declared schema are different").should("exist");` and remove the rest

Reply:
Agreed, I think it's OK to just check for "Detected schema and declared schema are different"